### ISSUE TITLE
Criterion for stopping training task

### ISSUE THREAD

**Issue by tf79** [Mri-monitoring/Mri-app#6]
Some trainings just do not converge and give nice results. Depending on the net size and hardware training might take a long time and often it's very easy in the beginning to see if a net is converging or not.

How about a criterion to automatically stop a training task i.e. if it's not converging after x iterations.
-  Maybe taking the mean over some iterations and using a threshold value for accuracy. 
- Or (more complex) using hyperparameter optimizer, i.e. Bayesian optimization (SMAC, TPE or [this](http://aad.informatik.uni-freiburg.de/papers/15-IJCAI-Extrapolation_of_Learning_Curves.pdf) )
